rotkehlchen/interfaces.py

```diff
@@ -0,0 +1,63 @@
+import abc
+from typing import Optional
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.types import Price, Timestamp
+
+
+class HistoricalPriceOracleInterface(metaclass=abc.ABCMeta):
+    """Query prices for certain timestamps. Oracle could be rate limited"""
+
+    def __init__(self, oracle_name: str):
+        self.oracle_name = oracle_name
+
+    @abc.abstractmethod
+    def rate_limited_in_last(
+            self,
+            seconds: Optional[int] = None,
+    ) -> bool:
+        ...
+
+    @abc.abstractmethod
+    def can_query_history(
+            self,
+            from_asset: Asset,
+            to_asset: Asset,
+            timestamp: Timestamp,
+            seconds: Optional[int] = None,
+    ) -> bool:
+        """Checks if it's okay to query historical price"""
+        ...
+
+    @abc.abstractmethod
+    def query_historical_price(
+            self,
+            from_asset: Asset,
+            to_asset: Asset,
+            timestamp: Timestamp,
+    ) -> Price:
+        ...
+
+
+class CurrentPriceOracleInterface(metaclass=abc.ABCMeta):
+    """
+    Interface for oracles able to query current price. Oracle could be rate limited
+    """
+
```

Comment:
Why is there no __init__() here to set oracle_name? I don't understand why one interface has a getter method which is a no-op apart from returning a constant string and the other has a ctor argument.